app/bundles/LeadBundle/Form/Type/CampaignActionAnonymizeUserDataType.php

```diff
@@ -0,0 +1,150 @@
+            ]
+        );
+
+        $choicesToDelete = $this->getFieldChoices();
+        $builder->add(
+            'fieldsToDelete',
+            FieldListType::class,
+            [
+                'label'       => 'mautic.lead.lead.events.delete_user_data',
+                'choices'     => $choicesToDelete,
+                'constraints' => [$this->checkFieldsSimilarity()],
+                'data'        => $options['data']['fieldsToDelete'] ?? self::DEFAULT_VALUES_TO_DELETE,
+            ]
+        );
+
+        // Add a text at the end of the form
+        $builder->add(
+            'customText',
+            TextType::class,
+            [
+                'label'      => $this->translator->trans('mautic.campaign.lead.action_anonymizeuserdata.alert.auditlog'), // No label for the text
+                'label_attr' => ['class' => 'text-muted'], // Optional: Add a custom CSS class to the label
+                'data'       => $this->translator->trans('mautic.campaign.lead.action_anonymizeuserdata.alert.auditlog'),
+                'mapped'     => false, // Not mapped to any entity field
+                'required'   => false, // Not required
+                'attr'       => [
+                    'readonly' => true, // Make it read-only
+                    'class'    => 'custom-text-class', // Optional: Add a custom CSS class
+                    'style'    => 'display: none;', // Optional: Add custom styles
+                ],
+            ]
+        );
+    }
+
+    /**
+     * @return array<string, int>
+     */
+    private function getFieldChoices(bool $checkIsUniqueField=true): array
+    {
+        $findBy['type'] = self::FIELD_TYPE_ALLOWED;
+        if ($checkIsUniqueField) {
+            $findBy['isUniqueIdentifer'] = false;
+        }
+        $leadFields    = $this->fieldModel->getRepository()->findBy($findBy);
+        $choices       = [];
+
+        foreach ($leadFields as $field) {
+            $choices[$field->getLabel()] = $field->getId();
+        }
+
+        return $choices;
+    }
+
+    public function getBlockPrefix(): string
+    {
+        return 'lead_action_anonymizeuserdata';
+    }
+
+    private function checkFieldsSimilarity(): Callback
+    {
+        return new Callback(
+            function ($validateMe, ExecutionContextInterface $context): void {
+                /** @var Integration $data */
+                $data = $context->getRoot()->getData();
+
+                if (
+                    !isset($data['properties']['fieldsToDelete'], $data['properties']['fieldsToAnonymize'])
+                    || (empty($data['properties']['fieldsToDelete']) && empty($data['properties']['fieldsToAnonymize']))
+                ) {
+                    $context->buildViolation('mautic.lead.lead.events.error.empty_fields')
+                        ->addViolation();
+                    $data['properties']['fieldsToDelete']    = [];
+                    $data['properties']['fieldsToAnonymize'] = [];
```

Comment:
Do you think this is needed? As during validation constraint, we should not alter the values.